apps/condo/domains/acquiring/hooks/usePaymentsTableColumns.ts

```diff
@@ -0,0 +1,85 @@
+import { getDateRender, getMoneyRender, getTextRender } from '@condo/domains/common/components/Table/Renders'
+import { getSorterMap, parseQuery } from '@condo/domains/common/utils/tables.utils'
+import { useIntl } from '@core/next/intl'
+import { get } from 'lodash'
+import { useRouter } from 'next/router'
+import { useMemo } from 'react'
+
+export function usePaymentsTableColumns (currencyCode: string): Record<string, unknown>[] {
+    const intl = useIntl()
+    const router = useRouter()
+
+    const addressTitle = intl.formatMessage({ id: 'field.Address' })
+    const dateTitle = intl.formatMessage({ id: 'Date' })
+    const accountTitle = intl.formatMessage({ id: 'field.AccountNumberShort' })
+    const unitNameTitle = intl.formatMessage({ id: 'field.FlatNumber' })
+    const typeTitle = intl.formatMessage({ id: 'PaymentType' })
+    const transactionTitle = intl.formatMessage({ id: 'Transaction' })
+    const paymentAmountTitle = intl.formatMessage({ id: 'PaymentAmount' })
+
+    const { filters, sorters } = parseQuery(router.query)
+    const sorterMap = getSorterMap(sorters)
+
+    return useMemo(() => {
+        let search = get(filters, 'search')
+        search = Array.isArray(search) ? null : search
+
+        const columns = {
+            date: {
+                title: dateTitle,
+                key: 'advancedAt',
+                dataIndex: ['advancedAt'],
+                sorter: true,
+                width: '112px',
+                render: getDateRender(intl, String(search)),
+            },
+            account: {
+                title: accountTitle,
+                key: 'accountNumber',
+                dataIndex: 'accountNumber',
+                width: '120px',
+                render: getTextRender(String(search)),
+            },
+            address: {
+                title: addressTitle,
+                key: 'address',
+                dataIndex: ['receipt', 'property', 'address'],
```

Comment:
What if receipt was deleted? I guess then we should display frozenReceipt properties as a fallback...

Logic here are:
1. `row` is raw object
2. Then subObject is taken by dataIndex...in our case `row.receipt.property.address`
3. subObject is passed to `render` function...so it's like `getTextRender(search)(row.receipt.property.address)`

What we can do instead is specify [] or undefinded as dataIndex and then overwrite render:
```
// Can move it to const above columns
const stringSeach = getTextRender(String(search))
render: (obj) => stringSearch(get(obj, ['receipt, ....], obj.frozenReceipt....))
```

Reply:
You right. Thanks for detailed explanation!
Fixed.